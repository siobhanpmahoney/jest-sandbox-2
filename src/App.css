.App {
  text-align: center;
}

* {
  font-family: "Avenir";

  /* position: relative; */
}

html, body {
  width: 100vw;
  /* height: 100%; */
  position: relative;

}

.App {
  height: 80vh;
  width: 90%;
  position: absolute;
  text-align: center;

    text-align: center;

  margin: 0px 12px;

}

#page-wrapper {
  /* border: #346EFF 2px solid; */
  border: #FFD57A 2px solid;
  height: 90%;
  width:96%;
  display: inline-flex;
  flex-flow: wrap row;
  justify-content: center;
  align-content: center;
  align-self: center;
  vertical-align: center;
  position: relative;
}

.day {
  background: linear-gradient(120deg, #89f7fe 0%, #66a6ff 100%);

  /* background-image: linear-gradient(120deg, #a1c4fd 0%, #c2e9fb 100%); */
  /* background-image: linear-gradient(60deg, #abecd6 0%, #fbed96 100%); */
  /* background-image: linear-gradient(120deg, #f6d365 0%, #fda085 100%); */
}

.night {
    border: #DBE6F1 solid 2px !important;
  background-image: linear-gradient(to top, #1e3c72 0%, #1e3c72 1%, #2a5298 100%);
  /* background: linear-gradient(-20deg, #2b5876 0%, #4e4376 100%); */
}

button.switch {
  width: 80px;
  position: relative;
  height: 35px;
  align-items: center;
  display: flex;
  justify-content: center;
  background-color: white;
  border-radius: 6px;
  text-align: center;
  margin: 50px;
  font-family: "Haboro Sans"

}
