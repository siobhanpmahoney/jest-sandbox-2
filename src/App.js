import React from 'react';
import "./App.css"

class App extends React.Component {
  constructor(props) {
    super(props)

    this.state = {
      toggle: true
    }
  }

  onToggle = () => {
    this.setState({
      toggle: !this.state.toggle
    })
  }

  render() {
    return (
      <div className="App">
      <h1>Jest Sandbox</h1>


      <div id="page-wrapper" className={this.state.toggle ? "day" : "night" }>
          <button className="switch" onClick={this.onToggle}>Switch</button>
        </div>

      </div>
      )
  }
}

export default App;
