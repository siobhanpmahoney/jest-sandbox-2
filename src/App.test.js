import React from 'react';
import ReactDOM from 'react-dom';
import App from './App';
import {mount, configure, shallow} from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';
import toJson from 'enzyme-to-json';

configure({ adapter: new Adapter() });

it('renders without crashing', () => {
  const div = document.createElement('div');
  ReactDOM.render(<App />, div);
  ReactDOM.unmountComponentAtNode(div);
});

describe('<App /> shallow renders', () => {
  it('correctly shallow renders', () => {
    const wrapper = shallow(<App />)
    expect(toJson(wrapper)).toMatchSnapshot()
  })
})
