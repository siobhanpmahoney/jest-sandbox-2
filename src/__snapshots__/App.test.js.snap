// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<App /> shallow renders correctly shallow renders 1`] = `
<div
  className="App"
>
  <h1>
    Jest Sandbox
  </h1>
  <div
    className="day"
    id="page-wrapper"
  >
    <button
      className="switch"
      onClick={[Function]}
    >
      Switch
    </button>
  </div>
</div>
`;
